doc/dev/cephadm/design/storage_devices_and_osds.rst

```diff
@@ -0,0 +1,317 @@
+  ceph orch apply osd -i <json_file/yaml_file> [--dry-run] [--unmanaged=true]*  (Service Spec based approach)
+
+**GUI**:
+Implemented in the dashboard section “cluster.OSDs”.
+There is a button to create the OSDs, that presents a page dialog box to select the physical devices that are going to be part of the OSDs as main, WAL or Db devices.
+It Is very difficult to make a selection ( or to understand how to make the selection). This is even worse if your cluster has the same kind of devices, resulting in the weird thing that is not possible to create an OSD using only one storage device (because you cannot select it)
+The problem here is the UI has been designed to work with “drive groups” and not to work for the user. The “drive group” is an abstract concept that must be used only in the background. Users must not be aware of this concept.
+
+B. Proposed workflow
+--------------------
+
+**CLI and GUI**
+
+The utilization of “declarative” drive groups makes it very difficult to understand how to configure OSD’s and the implications. Also make difficult the implementation because the multiple possibilities and the big amount of different conditions that we can find in a production system makes very complex the right evaluation and use of a declarative description of the storage devices used.
+This results in unexpected situations. For example:
+* A disk cleaned can be reused automatically and without any warning for creating a new OSD.
+* New installed disks are used automatically for OSDs (without any warning)
+* Errors trying to recreate OSD’s in disk removed from the system.
+
+So there is an important thing to consider in order to simplify everything for the user and for the implementation:
+**Avoid the “declarative” use of the drive groups**
+
+**GUI**:
+
+The user should be able to define the set of physical disk devices that are going to be used to support OSD’s.
+This means to make simple things like create one OSD in a certain device, and also to define in an easy way how to create multiple OSD’s across multiple devices in different hosts.
+
+We should take into account different premises:
+
+We use only bluestore OSD’s, this means that in order to create an OSD we can decide between different strategies: consume only a single device for the OSD, use an additional device for the WAL, and/or use another different device for the DB.
+To split the different bluestore OSD data components between different devices only makes sense if the WAL/DB devices are faster than the main storage device.
+And the split of devices are always inside the same host, although the configuration will be applicable to other hosts with the same storage devices schema.
+
+A massive creation of OSDs in a production system can result in real disaster because rebalancing can affect negatively to the normal system performance.
+The same massive OSD creation in a cluster that is being installed for the first time probably is the desired behaviour.
```

Comment:
```suggestion
The same massive OSD creation in a cluster that is being installed for the first time probably is the desired behavior.
```

Reply:
> 

@votdev AFAIK ```behaviour``` is the British spelling of ```behavior``` so it's also acceptable & correct.

Reply:
```
$ find * -name '*.rst' | xargs grep behavior | wc -l
176
$ find * -name '*.rst' | xargs grep behaviour | wc -l
33
```

Looks like both is ok.